src/remote/utils.go

```diff
@@ -158,6 +159,21 @@ func (c *Client) retrieveLocalResults(target *core.BuildTarget, digest *pb.Diges
 	return nil, nil
 }
 
+func retrieveTargetMetadataFromCache(c *Client, target *core.BuildTarget, digest *pb.Digest) *core.BuildMetadata {
+	if c.state.Cache.Retrieve(target, c.localCacheKey(digest), []string{target.TargetBuildMetadataFileName()}) {
+		// TODO(jpoole): Retrieving the metadata file from the cache loads it into the targets output directory. This feels like a
```

Comment:
+1 it feels funky to have to write it to disk somewhere else.

Hard to know what a good answer is here. I sort of preferred the idea that it got returned from the function before rather than requiring reading from a file - does it work to say that the metadata file is always in the cache and therefore the cache can always return it?

Reply:
Ah, I've just realised that the cached "file" is sometimes a directory. The current directory cache is recursively symlinking all the files under that dir. I don't think this idea will fit particularly well with that.